Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,287 @@
+        }
+        if (args.Handled || args.Cancelled || args.Args.Target == null || !_cell.TryUseActivatableCharge(uid, user: args.User))
+            return;
+
+        _audio.PlayPvs(component.ScanningEndSound, args.Args.User);
+
+        UpdateScannedUser(uid, args.Args.User, args.Args.Target.Value, component);
+        args.Handled = true;
+    }
+
+    private void OpenUserInterface(EntityUid user, EntityUid analyzer)
+    {
+        if (!TryComp<ActorComponent>(user, out var actor) || !_uiSystem.TryGetUi(analyzer, PlantAnalyzerUiKey.Key, out var ui))
+            return;
+
+        _uiSystem.OpenUi(ui, actor.PlayerSession);
+    }
+
+    public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid target, PlantAnalyzerComponent? plantAnalyzer)
+    {
+        if (!Resolve(uid, ref plantAnalyzer))
+            return;
+        if (target == null || !_uiSystem.TryGetUi(uid, PlantAnalyzerUiKey.Key, out var ui))
+            return;
+
+        TryComp<PlantHolderComponent>(target, out var plantcomp);
+        TryComp<SeedComponent>(target, out var seedcomponent);
+
+        var state = default(PlantAnalyzerScannedSeedPlantInformation);
+        var seedData = default(SeedData);   //data of a unique Seed
+        var seedProtoId = default(SeedPrototype);   //data of base seed prototype
+
+        if (seedcomponent != null)
+        {
+            if (seedcomponent.Seed != null) //if unqiue seed
+            {
+                seedData = seedcomponent.Seed;
+                state = ObtainingGeneDataSeed(seedData, target, false);
+            }
+            else if (seedcomponent.SeedId != null && _prototypeManager.TryIndex(seedcomponent.SeedId, out SeedPrototype? protoSeed))    //getting the protoype seed
+            {
+                seedProtoId = protoSeed;
+                state = ObtainingGeneDataSeedProt(protoSeed, target);
+            }
+        }
+        else if (plantcomp != null) //where check if we poke the plantholder, it checks the plantholder seed
+        {
+            _plantHolder = plantcomp;
+            seedData = plantcomp.Seed;
+            if (seedData != null)
+            {
+                //if (plantcomp.Health > 10) plantcomp.Health -= 10;
+                state = ObtainingGeneDataSeed(seedData, target, true);  //seedData is a unique seed in a tray
+            }
+        }
+
+        if (state == null)
+            return;
+        OpenUserInterface(user, uid);
+        _uiSystem.SendUiMessage(ui, state);
+    }
+
+    /// <summary>
+    ///     Analysis of seed from prototype. Shows all information since they are the "known stock" of NT.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeedProt(SeedPrototype comp, EntityUid target)
+    {
+        var seedName = "\r\n  ";
+        seedName = seedName + Loc.GetString("plant-analyzer-window-label-trademark") + "\r\n    " + Loc.GetString(comp.DisplayName);
+
+        var seedChem = "\r\n   ";
```

Comment:
Is the carriage return needed?

Reply:
No, it isn't...that happens if one(in this case me) blindly copies old code. Removed all of them.